[oneTBB] task_handle related extensions (#1)
anton-potapov/oneAPI-spec: source/elements/oneTBB/source/task_scheduler/task_arena/this_task_arena_ns.rst

        The object returned by the functor cannot be a reference. ``std::reference_wrapper`` can be used instead.

.. cpp:function:: template<typename F> void enqueue(F&& f)
  

>     Enqueues a task into the ``task_arena`` currently used by the calling thread to process the specified functor, then immediately returns.

Reviewer: ```suggestion
    Enqueues a task into the ``task_arena`` currently used by the calling thread to process the specified functor, then returns immediately.
```

Reviewer: Does it enqueues a task into the ``task_arena`` to process the specified functor or is `task_arena`` currently used by the calling thread to process the specified functor?

>     The ``F`` type must meet the `Function Objects` requirements from the [function.objects] ISO C++ Standard section.

Reviewer: ```suggestion
    The ``F`` type must meet the `Function Objects` requirements described in the [function.objects] section of the ISO C++ standard.
```

Author: I have just copied this section as is from the other sections of this file. If you think it should be changed  - then let's change all instances.

Reviewer: Yes, please change if you can!

Author: https://github.com/oneapi-src/oneAPI-spec/pull/372

>     Behavior of this function is identical to ``template<typename F> void task_arena::enqueue(F&& f)`` applied to ``task_arena`` 

Reviewer: ```suggestion
    Behavior of this function is identical to ``template<typename F> void task_arena::enqueue(F&& f)`` applied to the ``task_arena`` 
```

    object constructed with ``attach`` parameter.     

.. cpp:function:: void enqueue(task_handle&& h)   
     
    Enqueues a task owned by ``h`` into the ``task_arena`` that is currently used by the calling thread.
    

>     Behavior of this function is identical to generic version (``template<typename F> void enqueue(F&& f)``) except the parameter type. 

Reviewer: ```suggestion
    The behavior of this function is identical to the generic version (``template<typename F> void enqueue(F&& f)``), except the parameter type. 
```

    .. note:: 

>     ``h`` should not be empty to avoid undefined behavior.

Reviewer: ```suggestion
    ``h`` should not be empty to avoid an undefined behavior.
```
